scripts/abins/io.py

```diff
@@ -81,9 +91,39 @@ def _valid_advanced_parameters(self):
         :returns: True if they are the same, otherwise False
 
         """
+        from abins.parameters import non_performance_parameters as current_advanced_parameters
+
         previous_advanced_parameters = self.load(list_of_attributes=["advanced_parameters"])
-        return (abins.parameters.non_performance_parameters
-                == json.loads(previous_advanced_parameters["attributes"]["advanced_parameters"]))
+        previous_advanced_parameters = json.loads(
+            previous_advanced_parameters["attributes"]["advanced_parameters"])
+
+        diff = {key: (value, previous_advanced_parameters.get(key, None))
+                for key, value in current_advanced_parameters.items()
+                if previous_advanced_parameters.get(key, None) != value}
+        diff.update({key: (None, value)
+                     for key, value in previous_advanced_parameters.items()
+                     if key not in current_advanced_parameters})
+
+        sections = ('instruments', 'sampling', 'hdf_groups')
+
+        for section in sections:
+            if section in diff:
+                logger.information(f"Differences in Abins {section} parameters:")
+                new_group, old_group = diff[section]
+                for key in new_group:
+                    if key not in old_group:
+                        old_group[key] = '<key not present>'
+
+                    if new_group[key] != old_group[key]:
```

Comment:
If it's an `elif` then the `<key not present>` will not be visible because the logging block gets skipped in those cases. So we would need to add redundant logging lines to the `if` block.